ocamltest/builtin_variables.ml

```diff
@@ -146,3 +146,60 @@ let _ = List.iter Variables.register_variable
     test_fail;
     timeout;
   ]
+
+  (* Definition of builtin functions available for use *)
+
+(* The functions are listed in alphabetical order *)
+
+let bppm_decode_fun (arg: string) =
+  let l = String.length arg in
+  let b = Buffer.create 100 in
+  let rec loop i =
+    if i >= l then ()
+    else begin
+      let c = arg.[i] in
+      if c = '%' then begin
+        if (i + 1) >= l then ()
+        else
+          let nc = arg.[i+1] in
+          match nc with
+          | '#' -> Buffer.add_char b '%'; loop (i+2)
+          | '+' -> Buffer.add_char b '='; loop (i+2)
+          | '.' -> Buffer.add_char b ':'; loop (i+2)
+          | _ -> ()
+      end else begin
+        Buffer.add_char b c;
+        loop (i+1)
+      end
+    end
+  in loop 0;
+  let res = Buffer.contents b in
+  res
```

Comment:
It is wrong to duplicate logic to decode/encode BUILD_PATH_PREFIX_MAP inside the ocamltest codebase. Why are we not using the Build_path_prefix_map utility module here?